build/builder.go

```diff
@@ -505,15 +505,44 @@ func (o *Options) FindAllShards() []string {
 
 // IgnoreSizeMax determines whether the max size should be ignored.
 func (o *Options) IgnoreSizeMax(name string) bool {
+	matched := false
 	for _, pattern := range o.LargeFiles {
 		pattern = strings.TrimSpace(pattern)
-		m, _ := doublestar.PathMatch(pattern, name)
+		negated, validatedPattern := checkIsNegatePattern(pattern)
+
+		m, _ := doublestar.PathMatch(validatedPattern, name)
 		if m {
-			return true
+			// match against a negated LargeFiles pattern overrides all non-exclude matches
+			if negated {
+				return false
```

Comment:
I think it would be clearer logic if `matched = false` was done here. Then the logic is clear, the last matching pattern is the one that applies. Additionally, you can then go through the list backwards and return as soon as you find the first matching pattern. This is the usual logic you find in pattern lists.

Reply:
Thanks @keegancsmith, I wasn't familiar with the gitignore approach and was intending for any negate to override all other matches, regardless of order of the list. This is also why I was thinking about the bucket approach, since I didn't take order into account.

I've modified to be consistent with gitignore and will push changes after updating tests.